cmd/secrets/secret_helper.go

```diff
@@ -53,105 +81,112 @@ type secretsRequest struct {
 	Secrets []string `json:"secrets"`
 }
 
-// ReadSecrets implements a secrets reader from a directory/mount
-func readSecrets(r io.Reader, w io.Writer, dir string) error {
-	in, err := ioutil.ReadAll(r)
+type backend interface {
+	// secretID has a different format on each secret backend. In the file
+	// backend, it's a path, whereas in the kubernetes secret one it has the
+	// following format: "namespace/name/key".
+	get(secretID string) s.Secret
+}
+
+func decodeSecrets(r io.Reader, w io.Writer, dir string, usePrefixes bool, selector *backendSelector) error {
+	inputSecrets, err := parseInputSecrets(r)
 	if err != nil {
 		return err
 	}
 
+	var decodedSecrets map[string]s.Secret
+	if usePrefixes {
+		decodedSecrets = readSecretsUsingPrefixes(inputSecrets, selector)
+	} else {
+		decodedSecrets = readSecretsFromFile(inputSecrets, dir)
+	}
+
+	return writeDecodedSecrets(w, decodedSecrets)
+}
```

Comment:
nit: not a blocker but this can be a bit simplified:
```suggestion
	if usePrefixes {
		return writeDecodedSecrets(w, readSecretsUsingPrefixes(inputSecrets, selector))
	}
	
	return writeDecodedSecrets(w, readSecretsFromFile(inputSecrets, dir))
```

Reply:
Changed :+1: